php/class-media.php

```diff
@@ -1460,6 +1460,16 @@ private function create_attachment( $asset, $public_id ) {
 			$alt_text = wp_strip_all_tags( $asset['meta']['alt'] );
 			$this->update_post_meta( $attachment_id, '_wp_attachment_image_alt', $alt_text );
 		}
+		// Capture the Caption Text.
+		if ( ! empty( $asset['meta']['caption'] ) ) {
+			$caption = wp_strip_all_tags( $asset['meta']['caption'] );
+			wp_update_post(
+				array(
+					'ID'           => $asset['attachment_id'],
+					'post_excerpt' => $caption,
+				)
+			);
```

Comment:
@pereirinha - Do you not think this would be cleaner if you add the `post_excerpt` to the `$post_args` before creating the attachment and keep it in a single call?

Reply:
Great catch @DavidCramer. You're absolutely right.
I pushed an update.